src/structure/grid.ts

```diff
@@ -153,10 +158,12 @@ export class ViewersGrid {
         element: string | HTMLElement,
         indexer: EnvironmentIndexer,
         structures: Structure[] | UserStructure[],
+        properties: { [name: string]: Property },
```

Comment:
I think we should only store per-atom properties here, and stored them directly as numeric values:

```suggestion
        properties: {
            [name: string]: number[][] // array of per-atom values, one array per structure.
        },
```

This would mean doing the filtering you are currently doing in `_getAtomProperties` here instead, and only keeping what will be used later.